### DISCUSSION TITLE
Searching multiple tables.

### DISCUSSION THREAD

**Discussion by WadeDrift** [prisma/prisma#9487] [Q&A] [answered]
Sorry if this has been answered before, maybe someone can point me in the right direction. I couldn't find it searching the Q&A. I am fairly new to the backend process and seem to be a little confused on searching with Prisma. I am using PostgreSQL in a react app. There is a part where I want to search a table and pull all of the items that may match the search query. Right now I am using the contains approach, but I feel that would be inefficient when there are thousands of entries. Any help would be greatly appreciated. Here is the structure I have been playing around with: 

```
return prismaClient.application.findMany({
        where: {
          paid: true,

          //     refNumber: {
          //       search: searchText,
          //     },

          //     applicant: {
          //       firstName: { search: searchText },
          //     },

          OR: [
            {
              refNumber: {
                contains: searchText,
              },
            },
            { applicant: { firstName: { contains: searchText } } },
          ],
        },
      });
```

My end goal would be to have an item like:
```
     {
        firstName: 'Jon',
        lastName: 'Smith',
        id: '12345abcde'
     }
     {
        firstName: 'Mary',
        lastName: 'Smith',
        id: '6789abc'
     }
```
And then I could search '45abcd' (returns 1st), 'Smith' (returns both), 'smith' (returns both), 'SMTH' (returns both), 'abc Smith' (returns both).

Again any help would be greatly appreciated. Backend work is still new to me. Thanks in advance.

**Comment by ryands17** [comment, marked as answer]
You can't search for multiple values inside the same string, that wouldn't work. In the above example, searching for 'abc Smith', you can't find both by `id` and `lastName` with the same string.

You can use `contains` specifying a single value to search along with [case insensitive filtering](https://www.prisma.io/docs/concepts/components/prisma-client/case-sensitivity#postgresql-provider) like this:

```ts
const users = await prisma.user.findMany({
  where: {
  OR: [
    email: {
      contains: searchText,
      mode: 'insensitive',
    },
    firstName: {
      contains: searchText,
      mode: 'insensitive',
    },
    lastName: {
      contains: searchText,
      mode: 'insensitive',
    },
  ]
  },
})
```

This will give results when you search for '45abcd', 'Smith', and 'SMITH'.

**Reply by WadeDrift** [reply]
Thank you for your help. I definitely implement that into it.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
